### PR TITLE
[DOCS] Adds cohere service example to the inference API tutorial

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds an inference API example that uses the `cohere` service to the Inference API tutorial. The PR reworks the page to use tabular widgets in the instructions which make it possible to cover both the `cohere` and the `openai` service cases on the same page. The widgets are extendable.

### Preview

[Semantic search with the inference API](https://elasticsearch_bk_105904.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/semantic-search-inference.html)

### CODE REVIEW

File: docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  }
+}
+--------------------------------------------------
+<1> The name of the field to contain the generated tokens. It must be refrenced
+in the {infer} pipeline configuration in the next step.
+<2> The field to contain the tokens is a `dense_vector` field.
+<3> The output dimensions of the model. Find this value in the
+https://docs.cohere.com/reference/embed[Cohere documentation] of the model you
+use.
+<4> The name of the field from which to create the sparse vector representation.
```
Comment (jonathan-buttner) [inline docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc:27]:
```suggestion
<4> The name of the field from which to create the dense vector representation.
```

File: docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+in the {infer} pipeline configuration in the next step.
+<2> The field to contain the tokens is a `dense_vector` field.
+<3> The output dimensions of the model. Find this value in the
+https://platform.openai.com/docs/guides/embeddings/embedding-models[OpenAI documentation]
+of the model you use.
+<4> The faster` dot_product` function can be used to calculate similarity
+because OpenAI embeddings are normalised to unit length. You can check the
+https://platform.openai.com/docs/guides/embeddings/which-distance-function-should-i-use[OpenAI docs]
+about which similarity function to use.
+<5> The name of the field from which to create the sparse vector representation.
```
Comment (jonathan-buttner) [inline docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc:66]:
```suggestion
<5> The name of the field from which to create the dense vector representation.
```

File: docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+[source,console]
+------------------------------------------------------------
+PUT _inference/text_embedding/cohere_embeddings <1>
+{
+    "service": "cohere",
+    "service_settings": {
+        "api_key": "<api_key>", <2>
+        "model": "embed-english-light-v3.0" <3>
+    },
```
Comment (jonathan-buttner) [inline docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc:11]: How about we also add the field `embedding_type` to the example as well. Just showing that users can set it to `float` or `int8`.

File: docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+[source,console]
+------------------------------------------------------------
+PUT _inference/text_embedding/openai_embeddings <1>
+{
+    "service": "openai",
+    "service_settings": {
+        "api_key": "<api_key>" <2>
+    },
+    "task_settings": {
+       "model": "text-embedding-ada-002" <3>
```
Comment (jonathan-buttner) [inline docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc:40]: Unrelated to the cohere stuff but we decided to move this to the `service_settings`. We also renamed it to `model_id`. So `"model_id": "text-embedding-ada-002"`

File: docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc [hunk 5]
```diff
@@ -0,0 +1,55 @@
+// tag::cohere[]
+
+[source,console]
+------------------------------------------------------------
+PUT _inference/text_embedding/cohere_embeddings <1>
+{
+    "service": "cohere",
+    "service_settings": {
+        "api_key": "<api_key>", <2>
+        "model": "embed-english-light-v3.0" <3>
```
Comment (jonathan-buttner) [inline docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc:10]: Let's switch `model` to `model_id`.

**Review by jonathan-buttner — COMMENTED** [review]
Looks great, left a few comments

**Comment by szabosteve** [comment]
@jonathan-buttner Thank you for your comments. I addressed them in my latest commits.

File: docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline.asciidoc [hunk 6]
```diff
@@ -18,7 +18,7 @@ PUT _ingest/pipeline/cohere_embeddings
   "processors": [
     {
       "inference": {
-        "model_id": "cohere_embeddings", <1>
+        "inference_id": "cohere_embeddings", <1>
```
Comment (jonathan-buttner) [inline docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline.asciidoc:21]: Sorry my response was confusing. What I meant was that we refer to this field as the `inference_id` when actually performing the put request for the inference configuration. But the it is actually `model_id` in the ingest processor. So let's change this back to `model_id`.

File: docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline.asciidoc [hunk 7]
```diff
@@ -28,7 +28,7 @@ PUT _ingest/pipeline/cohere_embeddings
   ]
 }
 --------------------------------------------------
-<1> The name of the inference model you created by using the
+<1> The name of the inference configuration you created by using the
```
Comment (jonathan-buttner) [inline docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline.asciidoc:31]: 👍

**Review by jonathan-buttner — APPROVED** [review]
Sorry for the confusion, one more change and it looks good.

File: docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline.asciidoc [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+// tag::cohere[]
+
+[source,console]
+--------------------------------------------------
+PUT _ingest/pipeline/cohere_embeddings
+{
+  "processors": [
+    {
+      "inference": {
+        "inference_id": "cohere_embeddings", <1>
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline.asciidoc:21]:
```suggestion
        "model_id": "cohere_embeddings", <1>
```

File: docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline.asciidoc [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+// tag::openai[]
+
+[source,console]
+--------------------------------------------------
+PUT _ingest/pipeline/openai_embeddings
+{
+  "processors": [
+    {
+      "inference": {
+        "inference_id": "openai_embeddings", <1>
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline.asciidoc:48]:
```suggestion
        "model_id": "openai_embeddings", <1>
```

**Pull request merged**